src/instructorTabs/InstructorTabs.tsx

```diff
@@ -0,0 +1,60 @@
+import { useState, useEffect, useCallback, useContext } from 'react';
+import { useNavigate, useLocation } from 'react-router-dom';
+import { Tab, Tabs } from '@openedx/paragon';
+import { SlotContext, BaseSlotOperation, useSlotOperations } from '@openedx/frontend-base';
+
+export interface TabProps {
+  tab_id: string,
+  url: string,
+  title: string,
+}
+
+interface SlotWithElementOperation extends BaseSlotOperation {
+  element: React.ReactElement,
+}
+
+const InstructorTabs = () => {
+  const navigate = useNavigate();
+  const location = useLocation();
+  const { id: slotId } = useContext(SlotContext);
+  const widgets: SlotWithElementOperation[] = useSlotOperations(slotId) as SlotWithElementOperation[];
+
+  const [tabKey, setTabKey] = useState<string>('courseInfo');
+
+  const getActiveTabFromUrl = useCallback(() => {
+    const currentPath = location.pathname.split('/').pop() ?? '';
+    const activeTab = widgets.find((slot) => slot.element.props.url === currentPath)?.element;
+    return activeTab ? activeTab.props.tab_id : '';
+  }, [widgets, location.pathname]);
+
+  useEffect(() => {
+    setTabKey(getActiveTabFromUrl());
+  }, [getActiveTabFromUrl]);
```

Comment:
It's confusing that `getActiveTabFromUrl` is a function, but this effect is designed to run when the URL changes. I think this can be done without `useEffect` or `useCallback`, just using react-router hooks directly and computing derived data every time (no need to memoize). See above.